File: scripts/patch-clay-flint.js
// Patch pebble-clay dist.zip to support 'flint' (Pebble 2 Duo).
// Flint is hardware-identical to diorite — we add it as a copy.
// The Pebble SDK extracts dist.zip at build time, so we must patch the zip itself.
var cp   = require('child_process');
var path = require('path');
var fs   = require('fs');
var os   = require('os');

var distZip = path.join(__dirname, '..', 'node_modules', 'pebble-clay', 'dist.zip');

if (!fs.existsSync(distZip)) {
  console.log('pebble-clay dist.zip not found — skipping flint patch');
  process.exit(0);
}

// Check if already patched
var listing = cp.execSync('unzip -l "' + distZip + '"').toString();
if (listing.indexOf('flint') !== -1) {
  console.log('pebble-clay: flint already in dist.zip');
  process.exit(0);
}

var tmpDir = path.join(os.tmpdir(), 'clay-flint-' + process.pid);
try {
  cp.execSync('mkdir -p "' + tmpDir + '"');
  cp.execSync('unzip -q "' + distZip + '" -d "' + tmpDir + '"');

  var incDiorite = path.join(tmpDir, 'include', 'pebble-clay', 'diorite');
  var incFlint   = path.join(tmpDir, 'include', 'pebble-clay', 'flint');
  var binDiorite = path.join(tmpDir, 'binaries', 'diorite');
  var binFlint   = path.join(tmpDir, 'binaries', 'flint');

  if (!fs.existsSync(incFlint))  cp.execSync('cp -r "' + incDiorite + '" "' + incFlint + '"');
  if (!fs.existsSync(binFlint))  cp.execSync('cp -r "' + binDiorite + '" "' + binFlint + '"');

  // Repack into original zip path
  cp.execSync('cd "' + tmpDir + '" && zip -qr "' + distZip + '" .');
  console.log('pebble-clay: patched dist.zip with flint platform support');
} catch (e) {
  console.error('pebble-clay flint patch failed: ' + e.message);
  process.exit(1);
} finally {
  try { cp.execSync('rm -rf "' + tmpDir + '"'); } catch (e) {}
}

File: scripts/patch-clay-flint.py
"""Patch pebble-clay dist.zip to add 'flint' platform support (Pebble 2 Duo).
Flint is hardware-identical to diorite — we copy its entries into the zip.
The Pebble SDK extracts dist.zip at build time, so we patch the zip itself.
"""
import os, sys, shutil, tempfile, zipfile

dist_zip = os.path.join(os.path.dirname(__file__), '..', 'node_modules', 'pebble-clay', 'dist.zip')
dist_zip = os.path.normpath(dist_zip)

if not os.path.exists(dist_zip):
    print('pebble-clay dist.zip not found — skipping flint patch')
    sys.exit(0)

with zipfile.ZipFile(dist_zip, 'r') as z:
    names = z.namelist()

if any('flint' in n for n in names):
    print('pebble-clay: flint already in dist.zip')
    sys.exit(0)

tmpdir = tempfile.mkdtemp(prefix='clay-flint-')
try:
    with zipfile.ZipFile(dist_zip, 'r') as z:
        z.extractall(tmpdir)

    inc_src = os.path.join(tmpdir, 'include', 'pebble-clay', 'diorite')
    inc_dst = os.path.join(tmpdir, 'include', 'pebble-clay', 'flint')
    bin_src = os.path.join(tmpdir, 'binaries', 'diorite')
    bin_dst = os.path.join(tmpdir, 'binaries', 'flint')

    if not os.path.exists(inc_dst):
        shutil.copytree(inc_src, inc_dst)
    if not os.path.exists(bin_dst):
        shutil.copytree(bin_src, bin_dst)

    with zipfile.ZipFile(dist_zip, 'w', zipfile.ZIP_DEFLATED) as z:
        for root, dirs, files in os.walk(tmpdir):
            for f in files:
                full = os.path.join(root, f)
                arc  = os.path.relpath(full, tmpdir)
                z.write(full, arc)

    print('pebble-clay: patched dist.zip with flint platform support')
finally:
    shutil.rmtree(tmpdir, ignore_errors=True)
